packages/filebrowser/src/listing.ts

```diff
@@ -1987,13 +2006,20 @@ export namespace DirListing {
       node.appendChild(name);
       node.appendChild(narrow);
       node.appendChild(modified);
+      node.appendChild(fileSize);
```

Comment:
:thinking: performance-wise the approach taken by checkboxes seems better, maintenance-wise the simpler the better. Any benefit of not rendering these will only slightly increase the maximal bearable size of the directory before it starts lagging and we could save much more by lazy-rendering and using content-visibility CSS - so I think it is fine to proceed as is.

Reply:
Right, while working on this it looked like the file browser could benefit from some refactoring especially in how the hidden columns are handled.

Since I'm not sure we will want to add more columns in the near future I would also say it's fine to leave it as is for now to not delay the merge before the 4.0 feature freeze.

We can open an issue to track further improvements, which could happen during the beta and rc cycles if time permits.